backend/src/email/routes/tests/email-transactional.routes.test.ts

```diff
@@ -55,179 +89,626 @@ describe('POST /transactional/email/send', () => {
         reply_to: user.email,
       })
 
     expect(res.status).toBe(202)
     expect(mockSendEmail).toBeCalledTimes(1)
+    const transactionalEmail = await EmailMessageTransactional.findOne({
+      where: { userId: userId.toString() },
+    })
+    expect(transactionalEmail).not.toBeNull()
+    expect(transactionalEmail).toMatchObject({
+      recipient: validApiCall.recipient,
+      from: `Hello <${user.email}>`,
+      status: TransactionalEmailMessageStatus.Accepted,
+      errorCode: null,
+    })
+    expect(transactionalEmail?.params).toMatchObject({
+      subject: validApiCall.subject,
+      body: validApiCall.body,
+      from: `Hello <${user.email}>`,
+      replyTo: user.email,
+    })
   })
 
   test('Should thrown an error with invalid custom from address', async () => {
-    const mockSendEmail = jest
+    mockSendEmail = jest.spyOn(EmailService, 'sendEmail')
+    const res = await request(app)
+      .post(endpoint)
+      .set('Authorization', `Bearer ${apiKey}`)
+      .send({
+        ...validApiCall,
+        from: 'Hello <invalid@agency.gov.sg>',
+        reply_to: user.email,
+      })
+
+    expect(res.status).toBe(400)
+    expect(mockSendEmail).not.toBeCalled()
+  })
+
+  test('Should throw an error if email subject or body is empty after removing invalid HTML tags and correct error is saved in db', async () => {
+    mockSendEmail = jest.spyOn(EmailService, 'sendEmail')
+    const invalidHtmlTagsSubjectAndBody = {
+      subject: '\n\n\n',
+      body: '<script></script>',
+    }
+    const res = await request(app)
+      .post(endpoint)
+      .set('Authorization', `Bearer ${apiKey}`)
+      .send({
+        ...validApiCall,
+        subject: invalidHtmlTagsSubjectAndBody.subject,
+        body: invalidHtmlTagsSubjectAndBody.body,
+      })
+
+    expect(res.status).toBe(400)
+    expect(mockSendEmail).not.toBeCalled()
+
+    const transactionalEmail = await EmailMessageTransactional.findOne({
+      where: { userId: userId.toString() },
+    })
+    expect(transactionalEmail).not.toBeNull()
+    expect(transactionalEmail).toMatchObject({
+      recipient: validApiCall.recipient,
+      from: validApiCall.from,
+      status: TransactionalEmailMessageStatus.Unsent,
+    })
+    expect(transactionalEmail?.params).toMatchObject({
+      // NB sanitisation only occurs at sending step, doesn't affect saving in params
+      subject: invalidHtmlTagsSubjectAndBody.subject,
+      body: invalidHtmlTagsSubjectAndBody.body,
+      from: validApiCall.from,
+      replyTo: validApiCall.reply_to,
+    })
+    expect(transactionalEmail?.errorCode).toBe(EMPTY_MESSAGE_ERROR_CODE)
+  })
+
+  test('Should send email if subject and body are not empty after removing invalid HTML tags and metadata is saved correctly in db', async () => {
+    mockSendEmail = jest
       .spyOn(EmailService, 'sendEmail')
       .mockResolvedValue('message_id')
+    const invalidHtmlTagsSubjectAndBody = {
+      subject: 'HELLO',
```

Comment:
as I was writing this test, I realize that we're actually replacing `\n` and `\r\n` in the subject header with `<br />` as part of the sanitization process. I thought it's weird because there shouldn't be line breaks in the subject line at all. 

conversely, it seems like we're not doing this for the body? I'm not quite sure what's going on there (the comments suggest that it's doing this so that things render properly on the frontend, but I still don't fully understand), but I don't think it's worth the effort to fix.